File: clases 25-11-2025/Tipos de clases js/clases_abstractas/EstudianteReal.js
import Estudiante from './Estudiante.js';

export default class EstudianteReal extends Estudiante {
    constructor(nombre, grado) {
        if (nombre instanceof Estudiante) {
            super(nombre.nombre, nombre.grado);
        } else {
            super(nombre, grado);
        }
    }

    estudiar() {
        console.log(`${this.nombre} está estudiando...`);
    }
}

File: clases 25-11-2025/tipos de clases Java/clases_abstractas/EstudianteReal.java
public class EstudianteReal extends Estudiante {
    public EstudianteReal(String nombre, int grado) {
        super(nombre, grado);
    }

    public EstudianteReal(Estudiante otro) {
        super(otro);
    }

    @Override
    public void estudiar() {
        System.out.println(nombre + " está estudiando...");
    }
}